packages/appium/lib/extension/plugin-config.js

```diff
@@ -0,0 +1,105 @@
+// @ts-check
+
+import _ from 'lodash';
+import {ExtensionConfig} from './extension-config';
+import log from '../logger';
+import {PLUGIN_TYPE} from '../constants';
+
+/**
+ * @extends {ExtensionConfig<PluginType>}
+ */
+export class PluginConfig extends ExtensionConfig {
+  /**
+   * A mapping of `APPIUM_HOME` values to {@link PluginConfig} instances.
+   * Each `APPIUM_HOME` should only have one associated `PluginConfig` instance.
+   * @type {Record<string,PluginConfig>}
+   * @private
+   */
+  static _instances = {};
+
+  /**
+   * Call {@link PluginConfig.getInstance} instead.
+   *
+   * Just calls the superclass' constructor with the correct extension type
+   * @private
+   * @param {import('./manifest').Manifest} io - IO object
+   * @param {PluginConfigOptions} [opts]
+   */
+  constructor (io, {extData, logFn} = {}) {
+    super(PLUGIN_TYPE, io, logFn);
+
+    if (extData) {
+      this.validate(extData);
+    }
+  }
+
+  /**
+   *
+   * @param {import('./manifest').Manifest} io
+   * @param {PluginConfigOptions} [opts]
+   * @returns {PluginConfig}
+   */
+  static create (io, {extData, logFn} = {}) {
+    const instance = new PluginConfig(io, {logFn, extData});
+    PluginConfig._instances[io.appiumHome] = instance;
+    return instance;
+  }
+
+  /**
+   * Creates or gets an instance of {@link PluginConfig} based value of `appiumHome`
```

Comment:
hm, not sure this method creates anything, but IMHO it makes sense to combine it with `create` one

Reply:
yes, this is confusing and one of these functions should probably go away.